### PR TITLE
configured spring-boot-data-jpa and added flyway for migrations

### LINKED ISSUE #15: Make signature requests persistent
Signing requests do not survive a node reboot.

### PR DESCRIPTION
**Pull request by woutslakhorst**
Having a persistent DB helps in setting up a local network/test network, production network. Hell.... all networks

Fixes #15 

Signed-off-by: Wout Slakhorst <wout.slakhorst@nedap.com>

### CODE REVIEW

**Comment by woutslakhorst** [comment]
@stevenvegt Is it a hard requirement that approved certificate requests remain stored (for now). Since I changed logic so it deletes requests.

File: src/main/kotlin/nl/nuts/discovery/service/CertificateAndKeyService.kt [hunk 1]
```diff
@@ -19,50 +19,221 @@
...
+        val provider = Crypto.findProvider(signatureScheme.providerName)
+        val signer = ContentSignerBuilder.build(signatureScheme, issuerKeyPair.private, provider)
+        val builder = X509Utilities.createPartialCertificate(
+            certificateType,
+            issuer,
+            issuerKeyPair.public,
+            subject,
+            subjectPublicKey,
+            window,
+            null, // todo
```
Comment (reinkrul) [inline src/main/kotlin/nl/nuts/discovery/service/CertificateAndKeyService.kt]: TODO?

File: src/main/resources/db/migration/h2/V1__create_table_node.sql [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+CREATE TABLE node
+(
+    id   INT AUTO_INCREMENT PRIMARY KEY,
+    hash VARCHAR(64)  NOT NULL UNIQUE,
```
Comment (reinkrul) [inline src/main/resources/db/migration/h2/V1__create_table_node.sql]: You're querying on hash and name, but they're not indexed, do we need them?

File: src/main/resources/db/migration/h2/V3__create_table_certificate.sql [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+CREATE TABLE certificate
+(
+    id   INT AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(255) NOT NULL,
```
Comment (reinkrul) [inline src/main/resources/db/migration/h2/V3__create_table_certificate.sql]: same here: indexes?

File: src/main/resources/db/migration/h2/V2__create_table_certificate_request.sql [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+CREATE TABLE certificate_request
+(
+    id           INT AUTO_INCREMENT PRIMARY KEY,
+    name         VARCHAR(255) NOT NULL,
```
Comment (reinkrul) [inline src/main/resources/db/migration/h2/V2__create_table_certificate_request.sql]: same here: indexes?

File: src/main/resources/db/migration/h2/V1__create_table_node.sql [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+CREATE TABLE node
+(
+    id   INT AUTO_INCREMENT PRIMARY KEY,
```
Comment (reinkrul) [inline src/main/resources/db/migration/h2/V1__create_table_node.sql]: Why ints? (vs GUIDs)

File: src/main/kotlin/nl/nuts/discovery/service/CertificateAndKeyService.kt (hunk 1 above)
Comment (woutslakhorst) [reply src/main/kotlin/nl/nuts/discovery/service/CertificateAndKeyService.kt]: Nuts CA tree, constraints like in registry. I added some text to the todo

File: src/main/resources/db/migration/h2/V1__create_table_node.sql (hunk 2 above)
Comment (woutslakhorst) [reply src/main/resources/db/migration/h2/V1__create_table_node.sql]: Unique, so yes it has an index

File: src/main/resources/db/migration/h2/V3__create_table_certificate.sql (hunk 3 above)
Comment (woutslakhorst) [reply src/main/resources/db/migration/h2/V3__create_table_certificate.sql]: That one does need an index yes, Added

File: src/main/resources/db/migration/h2/V2__create_table_certificate_request.sql (hunk 4 above)
Comment (woutslakhorst) [reply src/main/resources/db/migration/h2/V2__create_table_certificate_request.sql]: added

File: src/main/resources/db/migration/h2/V1__create_table_node.sql (hunk 5 above)
Comment (woutslakhorst) [reply src/main/resources/db/migration/h2/V1__create_table_node.sql]: easiest support for generating

**Pull request merged**